ios/RNSScreenContainer.m

```diff
@@ -122,18 +111,48 @@ - (void)updateContainer
   // this is done to mimick the effect UINavigationController has when willMoveToWindow:nil is
   // triggered before the animation starts
   if (activeScreenAdded) {
-    for (RNSScreenView *screen in _reactSubviews) {
-      if (screen.active && [_activeScreens containsObject:screen]) {
-        [self detachScreen:screen];
-        // disable interactions for the duration of transition
-        screen.userInteractionEnabled = NO;
-      }
+    // Following the assumption that there are mostly two active screens at once,
+    // it's reasonable to assume that it's not allowed to add an additional one
+    // if there are already two of them
+    RCTAssert(_activeScreens.count < 2, @"It's allowed to have max two active screens");
+    RNSScreenView* onlyActiveScreen = [_activeScreens anyObject];
+    if (onlyActiveScreen != nil) {
+      // We're not using [self.detachScreen:] cause we don't
+      // want to actually call [view.removeFromSuperview] in order 
+      // to keep the gesture recognition continuous
+      [onlyActiveScreen.controller removeFromParentViewController];
+      [onlyActiveScreen.controller willMoveToParentViewController:nil];
+      [_activeScreens removeObject:onlyActiveScreen];
+      onlyActiveScreen.userInteractionEnabled = YES;
     }
 
-    // add new screens in order they are placed in subviews array
+    // add new screens in the order they were placed in subviews array
     for (RNSScreenView *screen in _reactSubviews) {
       if (screen.active) {
-        [self attachScreen:screen];
+        if (screen == onlyActiveScreen) {
+          [_controller addChildViewController:screen.controller];
+          [screen.controller didMoveToParentViewController:_controller];
+          [_activeScreens addObject:screen];
+        } else {
+          [_controller addChildViewController:screen.controller];
+          if (onlyActiveScreen == nil) {
+            [_controller.view addSubview:screen.controller.view];
```

Comment:
```suggestion
			 // the frame is already set at this moment because we adjust it in insertReactSubview. We don't
			 // want to update it here as react-driven animations may already run and hence changing the frame
			 // would result in visual glitches
            [_controller.view addSubview:screen.controller.view];
```